src/background/messenger.ts

```diff
@@ -131,7 +131,12 @@ export default class Messenger {
 
     reportChanges(data: ExtensionData) {
         if (this.changeListenerCount > 0 || isFirefox) {
-            chrome.runtime.sendMessage({name: 'background', type: 'changes', data});
+            const message: Message = {
```

Comment:
Don't declare variables if you end up using it once or don't modify it in a future reference.

Reply:
I declared this variable just to force a type annotation onto it. Is there a better way to do it?

This just "converts" the object into the `Message` type and does not error out the build:
```
            chrome.runtime.sendMessage({
                from: 'background',
                type: 'changes',
                data
            } as Message);
```

Reply:
Not really. As this kind of things just happens once it should be expected to be correct the first time. However in such hotpath the garbage collecter will like to recycle message variable instantly 🤷🏽

Reply:
I wish TypeScript compiler was aware of connection between `chrome.runtime.sendMessage` and `chrome.runtime.onMessage` and created a union type of all `onMessage` listeners and then forced it onto `sendMessage` arguments.

Reply:
Sounds a bit like Quantum Entanglement ;). Typescript allow for type constructors like `chrome.runtime.sendMessage<Message>()` But it seems like the `@types/chrome` doesn't have this implemented 🤷🏽

Reply:
> `chrome.runtime.sendMessage<Message>()`

That was the first thing I tried, but it did't work.

Reply:
The type definition needs explicit support for it. Opening up a issue/discussion https://github.com/DefinitelyTyped/DefinitelyTyped or even a PR for it is the most effective way.